### PR TITLE
fix: CADD while using --hgvs flag

### PR DESCRIPTION
**Pull request by diegomscoelho**
Jira: [Card](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-4859)

## Description

An user (https://github.com/Ensembl/ensembl-vep/issues/812) found out that while we are using VEP output from CADD is incomplete. This PR fix absence of CADD results while using --hgvs or --hgvsp

## Technical description

* When `hgvs_protein` function in `TranscriptVariationAllele.pm` tries to output hgvs_protein string, several alterations are made in $vfoa object. HGVS alterations might alter any --custom / --plugin outputs since these flags are launched after hgvsp output.
* It can be important to look up if `--hgvs` flag might alter other output that use` $vfoa`.

## Test

1) Try annotate VCF with `--hgvs` using or not this PR, use for example:

```sh
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  NA12878_58
chr1    78024345        .       C       CTAT    1777    .       SAP=26;ABP=0;MQM=60     GT:DP:AO        1/1:61:59
chr1    152681680       .       C       CAGCTCTGGGGGCTGCTGT     2922    .       SAP=21;ABP=0;MQM=59     GT:DP:AO        1/1:107:101
```

## Dependencies

If this PR is approved, https://github.com/Ensembl/ensembl-variation/pull/628 will be no longer necessary.

### CODE REVIEW

**Review by olaaustine — COMMENTED** [review]
Hi @diegomscoelho,
Tested with the test example given, and it works. 
Thank you!

File: modules/Bio/EnsEMBL/Variation/TranscriptVariationAllele.pm [hunk 1]
```diff
@@ -1724,7 +1726,8 @@ sub hgvs_protein {
   }
 
   ##### String formatting
-  return $self->_get_hgvs_protein_format($hgvs_notation);
+  my $hgvs_notation_output = $self->_get_hgvs_protein_format($hgvs_notation);
```
[the remark this answers is not in the document]
Comment (diegomscoelho) [reply modules/Bio/EnsEMBL/Variation/TranscriptVariationAllele.pm]: Done! Thanks for reviewing @olaaustine

**Review by olaaustine — APPROVED** [review]
All good. Thanks @diegomscoelho

**Comment by olaaustine** [comment]
Merged into release/108 and main.

**Comment by nuno-agostinho** [comment]
Cherry-picked to release/107.

**Pull request merged**